src/Microsoft.ComponentDetection.Detectors/npm/NpmLockfileDetectorBase.cs

```diff
@@ -176,6 +176,7 @@ private void ProcessIndividualPackageJTokens(ISingleFileComponentRecorder single
             var packageJsonToken = JToken.ReadFrom(reader);
             var enqueued = this.TryEnqueueFirstLevelDependencies(topLevelDependencies, packageJsonToken["dependencies"], dependencyLookup, skipValidation: skipValidation);
             enqueued = enqueued && this.TryEnqueueFirstLevelDependencies(topLevelDependencies, packageJsonToken["devDependencies"], dependencyLookup, skipValidation: skipValidation);
+            enqueued = enqueued && this.TryEnqueueFirstLevelDependencies(topLevelDependencies, packageJsonToken["optionalDependencies"], dependencyLookup, skipValidation: skipValidation);
```

Comment:
Since we are changing the dependency graph generated by the detector, we need to bump the version of all the detectors that use this lockfile base.

Reply:
fixed in
 https://github.com/microsoft/component-detection/pull/1030/commits/53fd28595087be4f75469600bdb7d7680f29c85e